File: backend/app/models/historique.py
from sqlalchemy import Column, Integer, String, Float, DateTime, Text, ForeignKey
from sqlalchemy.orm import relationship
from app.database import Base
from datetime import datetime
import enum


class TypeCargaison(str, enum.Enum):
    POCHE_SANG = "poche_sang"
    DEFIBRILLATEUR = "defibrillateur"
    MEDICAMENT = "medicament"
    PIECE_MECANIQUE = "piece_mecanique"
    FRAGILE = "fragile"
    PERISSABLE = "perissable"
    AUTRE = "autre"


class StatutMission(str, enum.Enum):
    EN_ATTENTE = "en_attente"
    EN_COURS = "en_cours"
    TERMINEE = "terminee"
    ANNULEE = "annulee"
    ECHEC = "echec"


class HistoriqueMission(Base):
    __tablename__ = "historique_missions"

    historique_id = Column(Integer, primary_key=True, index=True)
    mission_id = Column(Integer, ForeignKey("missions.mission_id"), nullable=False)
    drone_id = Column(Integer, ForeignKey("drones.drone_id"), nullable=False)
    date = Column(DateTime, default=datetime.utcnow)
    performance = Column(String(255), nullable=True)  # performance de la mission (réussie, retard, échec)
    commentaires = Column(Text, nullable=True)  # commentaires sur la mission

    # Relations
    mission = relationship("Mission", back_populates="historique")
    drone = relationship("Drone", back_populates="historique_missions")

File: frontend/lib/models/historique.dart
class HistoriqueMission {
  final int? historiqueId;
  final int missionId;
  final int droneId;
  final DateTime date;
  final String? performance;
  final String? commentaires;

  HistoriqueMission({
    this.historiqueId,
    required this.missionId,
    required this.droneId,
    required this.date,
    this.performance,
    this.commentaires,
  });

  Map<String, dynamic> toJson() {
    return {
      'mission_id': missionId,
      'drone_id': droneId,
      'date': date.toIso8601String(),
      if (performance != null) 'performance': performance,
      if (commentaires != null) 'commentaires': commentaires,
    };
  }

  factory HistoriqueMission.fromJson(Map<String, dynamic> json) {
    return HistoriqueMission(
      historiqueId: json['historique_id'],
      missionId: json['mission_id'],
      droneId: json['drone_id'],
      date: DateTime.parse(json['date']),
      performance: json['performance'],
      commentaires: json['commentaires'],
    );
  }
}